crates/veryl/src/cmd_check.rs

```diff
@@ -12,25 +12,38 @@ pub struct CmdCheck {
     opt: OptCheck,
 }
 
-#[derive(Error, Diagnostic, Debug, Default)]
+#[derive(Error, Diagnostic, Debug)]
 #[error("veryl check failed")]
 pub struct CheckError {
     #[related]
     pub related: Vec<AnalyzerError>,
+    error_count: u32,
+    error_count_limit: u32,
 }
 
 impl CheckError {
-    pub fn append(mut self, x: &mut Vec<AnalyzerError>) -> Self {
-        self.related.append(x);
+    pub fn new(error_count_limit: u32) -> Self {
+        Self {
+            related: Vec::new(),
+            error_count: 0,
+            error_count_limit,
+        }
+    }
+
+    pub fn append(mut self, x: &Vec<AnalyzerError>) -> Self {
+        for error in x {
```

Comment:
To avoid `clone`, you can use `drain` like below:

```rust
for error in x.drain(0..) {
}
```

`drain` takes `&mut Vec`, and the `error` can be directly added to `self.related` because it is moved from `x`.